### ISSUE TITLE
pip-sync can't update pip-tools

### ISSUE THREAD

**Issue by MinchinWeb** [jazzband/pip-tools#478]
Pip-tools can't update itself (at least on Windows). This is due to the fact that pip tries to delete the running script, which the OS won't allow while the script is running.

Two possible solutions:

1. Running the module directly (using python's `-m` switch that looks for a module's `__main__.py` file). This is what pip itself does. (i.e. to update pip, you run `python -m pip install pip --upgrade`)
2. blacklist pip-tools from updating itself and warn the user that they need to take care of this manually.

##### Environment Versions

1. OS Type: Windows 10x64
1. Python version: `3.6.1`
1. pip version:  `9.0.1`
1. pip-tools version: `1.8.0`

##### Steps to replicate

1. `pip install pip-tools==1.8.0` (or really any version but the current one)
2. `pip-tools > requirements.in`
3. `pip-compile` (this should generate a `requirements.txt` file that lists `pip-tools==1.8.1`)
4. `pip-sync`

##### Expected result

Not sure, but it shouldn't blow up

##### Actual result

```
> pip-sync requirements.txt
Collecting pip-tools==1.8.1
  Using cached pip_tools-1.8.1-py2.py3-none-any.whl
Requirement already satisfied: six in s:\tmp\venv2\lib\site-packages (from pip-tools==1.8.1)
Requirement already satisfied: first in s:\tmp\venv2\lib\site-packages (from pip-tools==1.8.1)
Requirement already satisfied: click>=6 in s:\tmp\venv2\lib\site-packages (from pip-tools==1.8.1)
Installing collected packages: pip-tools
  Found existing installation: pip-tools 1.8.0
    Uninstalling pip-tools-1.8.0:
Exception:
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\shutil.py", line 544, in move
    os.rename(src, real_dst)
OSError: [WinError 17] The system cannot move the file to a different disk drive: 's:\\documents\\code\\burst-energy-crm\\venv2\\scripts\\pip-sync.exe' -> 'C:\\Users\\William\\AppData\\Local\\Temp\\pip-ozje6gwh-uninstall\\documents\\code\\burst-energy-crm\\venv2\\scripts\\pip-sync.exe'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "s:\tmp\venv2\lib\site-packages\pip\basecommand.py", line 215, in main
    status = self.run(options, args)
  File "s:\tmp\venv2\lib\site-packages\pip\commands\install.py", line 342, in run
    prefix=options.prefix_path,
  File "s:\tmp\venv2\lib\site-packages\pip\req\req_set.py", line 778, in install
    requirement.uninstall(auto_confirm=True)
  File "s:\tmp\venv2\lib\site-packages\pip\req\req_install.py", line 754, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "s:\tmp\venv2\lib\site-packages\pip\req\req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "s:\tmp\venv2\lib\site-packages\pip\utils\__init__.py", line 267, in renames
    shutil.move(old, new)
  File "C:\Program Files\Python36\lib\shutil.py", line 559, in move
    os.unlink(src)
PermissionError: [WinError 5] Access is denied: 's:\\tmp\\venv2\\scripts\\pip-sync.exe'
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Program Files\Python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "S:\tmp\venv2\Scripts\pip-sync.exe\__main__.py", line 9, in <module>
  File "s:\tmp\venv2\lib\site-packages\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "s:\tmp\venv2\lib\site-packages\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "s:\tmp\venv2\lib\site-packages\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "s:\tmp\venv2\lib\site-packages\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "s:\tmp\venv2\lib\site-packages\piptools\scripts\sync.py", line 72, in cli
  File "s:\tmp\venv2\lib\site-packages\piptools\sync.py", line 157, in sync
  File "C:\Program Files\Python36\lib\subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['pip', 'install', 'pip-tools==1.8.1']'
returned non-zero exit status 2.
```

**Comment by davidovich** [comment]
This should be fixed when #461 is merged. Pip also went this way because windows is very keen on holding file handles more than we would deem necessary.

**Comment by davidovich** [comment]
Closing, will be fixed by the upcoming 1.9.0. In the meantime, you can try installing the master branch and use the `python -m piptools compile` invocation. Do not hesitate to reopen if need be.
